src/SearchImpl/Indexing/DistributedIndexingActivityQueue.cs

```diff
@@ -127,7 +129,14 @@ private static void Startup(int lastDatabaseId, int lastExecutedId, int[] gaps,
             IndexingActivityHistory.Reset();
         }
 
-        public static CompletionState GetCurrentCompletionState()
+        public static void ShutDown()
```

Comment:
What if all the classes would just implement IDisposable?

Reply:
Good idea. But these functions must be started / stopped in fixed order. I think Startup / Shutdown pair is more intuitive in this case.